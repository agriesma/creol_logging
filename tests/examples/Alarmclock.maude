load creol-interpreter .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op state : -> State [ctor] .
  eq state = {
    < "Alarmclock" : Class | Inh: noInh, Param: noVid, Att: "clock" |-> null,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null,
              Code: assign( "clock" ; int(0) ) ; return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "ccglab0" |-> null, "_" |->
              null, Code: assign( "clock" ; "+" ( "clock" :: int(1) ) ) ;
              call( "ccglab0" ; "this" ; "run" ; emp ) ; free( "ccglab0" ) ;
              return ( emp ) >,
            < "wakeme" : Method | Param: "n", Att: "n" |-> null,
              "alarmsetting" |-> null, "_" |-> null,
              Code: assign( "alarmsetting" ; "+" ( "clock" :: "n" ) ) ; await
              ">=" ( "clock" :: "alarmsetting" ) ; return ( emp ) >,
      Ocnt: 0 > } .
endm

