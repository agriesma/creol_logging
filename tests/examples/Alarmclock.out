==========================================
reduce in PROGRAM : state .
result State: {
< "Alarmclock" : Class | Inh: noInh, Param: noVid, Att: "clock" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: assign("clock" ;
    int(0)) ; return(emp) >, < "run" : Method | Param: noVid, Att: "_" |->
    null, "ccglab0" |-> null, Code: assign("clock" ; "+"("clock" :: int(1))) ;
    call("ccglab0" ; "this" ; "run" ; emp) ; free("ccglab0") ; return(emp) >, <
    "wakeme" : Method | Param: "n", Att: "_" |-> null, "alarmsetting" |-> null,
    "n" |-> null, Code: assign("alarmsetting" ; "+"("clock" :: "n")) ; await
    ">="("clock" :: "alarmsetting") ; return(emp) >,
  Ocnt: 0 >
}
Bye.
